OpenRA.Game/Network/ReplayConnection.cs

```diff
@@ -55,6 +59,14 @@ public ReplayConnection(string replayFilename)
 					var frame = BitConverter.ToInt32(packet, 0);
 					chunk.Packets.Add(Pair.New(client, packet));
 
+					if (!clientsFrame.ContainsKey(client))
+					{
+						clientIds.Add(client);
+						clientsFrame[client] = new List<int>();
+					}
+
+					clientsFrame[client].Add(frame);
```

Comment:
It looks like you could directly update ```lastClientsFrame``` here with the highest frame for each client, since below you're only picking out the highest frame from the whole list that you're putting into this ```clientsFrame``` dictionary.